src/main/java/org/opensearch/security/dlic/rest/api/ActionGroupsApiAction.java

```diff
@@ -92,8 +93,35 @@ public List<Route> routes() {
     }
 
     @Override
-    protected AbstractConfigurationValidator getValidator(final RestRequest request, BytesReference ref, Object... param) {
-        return new ActionGroupValidator(request, isSuperAdmin(), ref, this.settings, param);
+    protected RequestContentValidator createValidator(final Object... params) {
+        return RequestContentValidator.of(new RequestContentValidator.ValidationContext() {
+            @Override
+            public Object[] params() {
+                return params;
+            }
+
+            @Override
+            public Settings settings() {
+                return settings;
+            }
+
+            @Override
+            public Map<String, RequestContentValidator.DataType> allowedKeys() {
+                final ImmutableMap.Builder<String, DataType> allowedKeys = ImmutableMap.builder();
+                if (isSuperAdmin()) {
+                    allowedKeys.put("reserved", DataType.BOOLEAN);
+                }
+                allowedKeys.put("allowed_actions", DataType.ARRAY);
+                allowedKeys.put("description", DataType.STRING);
+                allowedKeys.put("type", DataType.STRING);
```

Comment:
String is overly generic here. Is there an enum for type (i.e. INDEX, CLUSTER) somewhere?

Reply:
Good point! There are no enums or constants except MutitenecyActionApi. I can extract all of them as enums, and some of them have the same name in the next small PR.